File: clients/Ruby/model/repair_action.rb
class RepairAction
    attr_accessor :target
    def initialize(target)
        @target = target
    end
    def self.read_from(stream)
        target = stream.read_int()
        RepairAction.new(target)
    end
    def write_to(stream)
        stream.write_int(@target)
    end
end

File: clients/Scala/src/main/scala/model/RepairAction.scala
package model

import util.StreamUtil

case class RepairAction(target: Int) {
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, target)
    }
}
object RepairAction {
    def readFrom(stream: java.io.InputStream): RepairAction = RepairAction(
        StreamUtil.readInt(stream)
        )
}
